lib/presentation/view/movie_list_view.dart

```diff
@@ -1,49 +1,61 @@
 import 'package:flutter/material.dart';
 
-import '../../domain/repository/movie_repository_interface.dart';
+import '../../core/util/constants/ui_constants.dart';
+import '../../data/model/data_state.dart';
+import '../bloc/movies_bloc.dart';
 import '../widget/movie_info.dart';
-import '../../data/repository/movie_repository.dart';
 import '../../domain/entity/movie.dart';
 
+
 class MovieListView extends StatefulWidget {
-  const MovieListView({super.key});
+  const MovieListView({
+    super.key,
+    required this.bloc,
+  });
+  final MoviesBloc bloc;
 
   @override
   State<MovieListView> createState() => _MovieListViewState();
 }
 
 class _MovieListViewState extends State<MovieListView> {
-  final IMovieRepository movieRepository = MovieRepositoryImp();
-  late Future<List<Movie>> movies;
   static const double dividerHeight = 20;
   static const String errorMessage = 'An error has occurred while loading data';
 
   @override
   void initState() {
     super.initState();
-    movies = movieRepository.getMovies();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    return FutureBuilder(
-      future: movies,
+    return StreamBuilder<DataState>(
+      initialData: widget.bloc.initialData,
+      stream: widget.bloc.movies,
       builder: (
         BuildContext context,
-        AsyncSnapshot<List<Movie>> snapshot,
+        AsyncSnapshot<DataState> snapshot,
```

Comment:
You are using the DataState that is defined on data on presentation layer, you need to have like a Event that is created base of the result of the UseCase that has the access of the DataState

Reply:
done